pyreadstat/pyreadstat.pyx

```diff
@@ -577,6 +580,51 @@ def read_file_in_chunks(read_function, file_path, chunksize=100000, offset=0, li
             offset += chunksize
             maxrow += chunksize
 
+def read_file_multiprocessing(read_function, file_path, num_processes=None, **kwargs):
+    """
+    Reads a file in parallel using multiprocessing.
+
+    Parameters
+    ----------
+        read_function : pyreadstat function
+            a pyreadstat reading function
+        file_path : string
+            path to the file to be read
+        num_processes : integer, optional
+            number of processes to spawn, by default the total number of cores
+        kwargs : dict, optional
+            any other keyword argument to pass to the read_function. 
+            row_limit and row_offset will be discarded if present as they are used internally.
+
+    Returns
+    -------
+        data_frame : pandas dataframe
+            a pandas data frame with the data
+        metadata :
+            object with metadata. Look at the documentation for more information.
+    """
+    if "row_offset" in kwargs:
+        _ = kwargs.pop("row_offset")
+
+    if "row_limit" in kwargs:
+        _ = kwargs.pop("row_limit")
+
+    if not num_processes:
+        num_processes = mp.cpu_count()
+    _, meta = read_function(file_path, metadataonly=True)
+    numrows = meta.number_rows
```

Comment:
ignoring row_offset and row_limit is not necessary AFAICT. Couldn't you support them using something like this **untested** code:
```python
import math  # <-- somewhere above

row_offset = kwargs.pop("row_offset", 0)
row_limit = kwargs.pop("row_limit", math.inf)
numrows = min(max(numrows - row_offset, 0), row_limit)
chunksize = numrows // num_processes + (1 if 0 < numrows % num_processes else 0)
offsets = [row_offset + indx * chunksize for indx in range(num_processes)] 
```
Note that `divs` is apparently not used except to compute `chunksize`, so you can simplify the code of `chunksize`.

Also, the rest of the code would be more readable (IMHO) if you spelled `jobs` like this:

```python
jobs = [(read_function, file_path, offset, chunksize, kwargs) for offset in offsets]
```

and changed the worker accordingly:

```python
def worker(inpt):
    read_function, path, row_offset, row_limit, kwargs = inpt
    df, meta = read_function(path, row_offset=row_offset, row_limit=row_limit, **kwargs)
    return df 
```
or even (if supporting Python < 3.3 is not required) using pool.starmap instead of map so the worker can be just:
```python
def worker(read_function, path, row_offset, row_limit, kwargs):
    df, meta = read_function(path, row_offset=row_offset, row_limit=row_limit, **kwargs)
    return df 
```

Reply:
Thanks for the suggestions! I'll look at them in detail.

Reply:
hey @gdementen, nice corrections! I haven't tested either but it looks good.

If you are interested you could send a pull request (against the dev branch, not against master) that I will approve, so that the commits are under your name. I will then test and this will come live on the next release (not immediately but when some more changes accumulate). If you like you could also correct the docstring of the function to reflect that row_limit and row_offset works. Regarding starmap, at the moment I am still building wheels for Python 2.7 so it should work there (never used starmap, no idea).

If you prefer not doing a pull request, no problem, I can do it later.

Reply:
I need the corrections on row_offset and row_limit in order to use this function from read in chunks, therefore I am moving forward to do all the changes. Since you didn't answer I assume it's OK. Thanks for the suggestions!

Reply:
Yes, of course I am ok with that. It's what I assumed you would do. If you prefer, I can do a PR about this tomorrow but since I am unable to easily test the patch anyway, you will probably have to modify it, so I am unsure it's worth waiting for a PR from me. Thanks for the offer though. I'll check tomorrow if you did it or not.

Reply:
Oops, I just noticed you did this already 😄. That's nice.

Reply:
thanks again, and feel free to send more suggestions or PRs (either way is fine) any time!